internal/aws/proxy/server.go

```diff
@@ -93,6 +93,11 @@ func NewServer(cfg *Config, logger *zap.Logger) (Server, error) {
 			req.URL.Host = awsURL.Host
 			req.Host = awsURL.Host
 
+			// Sig4 requires Content-Length to empty for empty bodies.
+			if req.Header.Get("Content-Length") == "0" {
+				req.Header.Set("Content-Length", "")
```

Comment:
is the correct fix to remove the header? the RFC seems to indicate so.

Reply:
Tested with removing it, works